metadata-db-app/src/cmr/metadata_db/data/oracle/providers.clj

```diff
@@ -79,35 +80,48 @@
 
 (defn save-provider
   [db provider]
-  (let [{:keys [provider-id short-name cmr-only small consortiums]} provider]
+  (let [{:keys [provider-id short-name cmr-only small consortiums metadata]} provider
+        metadata (if (some? metadata) (-> metadata pr-str cutil/string->gzip-bytes))]
     (j/insert! db
                :providers
-               ["provider_id" "short_name" "cmr_only" "small" "consortiums"]
-               [provider-id short-name (if cmr-only 1 0) (if small 1 0) consortiums])
-    (when (not small)
-      (ct/create-provider-concept-tables db provider))))
+               ["provider_id" "short_name" "cmr_only" "small" "consortiums" "metadata"]
+               [provider-id
+                short-name
+                (if cmr-only 1 0)
+                (if small 1 0)
+                consortiums
+                metadata
+                ])
+      (when (not small)
+        (ct/create-provider-concept-tables db provider))))
 
 (defn get-providers
+  "Get all providers but return it in the older minimal format"
   [db]
-  (map dbresult->provider
-       (j/query db ["SELECT * FROM providers"])))
+  (map dbresult->provider (j/query db ["SELECT * FROM providers"])))
 
 (defn get-provider
   [db provider-id]
+  ;; TODO: drop the short_name field and return it here as 'provider_id as short_name'
   (first (map dbresult->provider
               (j/query db
-                       [(str "SELECT provider_id, short_name, cmr_only, small, consortiums"
+                       [(str "SELECT provider_id, short_name, cmr_only, small, consortiums, metadata"
                              " FROM providers"
                              " WHERE provider_id = ?")
                         provider-id]))))
 
 (defn update-provider
-  [db {:keys [provider-id short-name cmr-only consortiums]}]
+  [db {:keys [provider-id short-name cmr-only consortiums metadata]}]
+  ;; Ignoring short-name as CMR will no longer support updates on this field as
+  ;; is is always the same as provider id. As of 2023-03-02, all production
+  ;; providers currently match.
```

Comment:
Should use the [deprecated](https://guide.clojure.style/#deprecated) docstyle.

Reply:
will this not mark the entire function as deprecated? I can just remove short-name and put something in a comment,

Reply:
`(defn update-provider
  "Update a provider in the database using the supplied provider-id allowing
   cmr_only, consortiums, and metadata to change. All other fields are not
   modified.

   Ignoring short-name as CMR will no longer support updates on this
   field as is is always the same as provider id. As of 2023-03-02, all
   production providers currently match."
  [db {:keys [provider-id cmr-only consortiums metadata]}]
  (j/update! db
             :providers
             {:cmr_only (if cmr-only 1 0)
              :consortiums consortiums
              :metadata (-> metadata
                            pr-str
                            cutil/string->gzip-bytes)}
             ["provider_id = ?" provider-id]))`